File: app/src/main/java/com/cool/eye/scan/decode/LuminanceSource.java
package com.cool.eye.scan.decode;

import android.graphics.Bitmap;

public abstract class LuminanceSource extends com.google.zxing.LuminanceSource {

  protected LuminanceSource(int width, int height) {
    super(width, height);
  }

  public abstract Bitmap renderCroppedGreyScaleBitmap();
}

File: app/src/main/java/com/eye/cool/scan/decode/source/LuminanceSource.kt
package com.eye.cool.scan.decode.source

import android.graphics.Bitmap
import com.google.zxing.LuminanceSource

internal abstract class LuminanceSource protected constructor(
    width: Int,
    height: Int
) : LuminanceSource(width, height) {
  abstract fun renderCroppedGreyScaleBitmap(): Bitmap?
}